core/sequencer/server.go

```diff
@@ -304,20 +291,56 @@ func (s *Server) DefineRevisions(ctx context.Context,
 	// TODO(#1047): If time since oldest queue item > max latency has elapsed, define batch.
 	// If count items >= min_batch, define batch.
 	if count >= in.MinBatch {
-		nextRev := int64(latestMapRoot.Revision) + 1
+		nextRev := resp.UnappliedBegin
+		resp.UnappliedEnd = nextRev + 1
 		if err := s.batcher.WriteBatchSources(ctx, in.DirectoryId, nextRev, meta); err != nil {
 			return nil, err
 		}
 		for _, source := range meta.Sources {
 			watermarkDefined.Set(float64(source.HighestExclusive),
 				in.DirectoryId, fmt.Sprintf("%v", source.LogId))
 		}
-		outstanding = append(outstanding, nextRev)
-
 	}
 	// TODO(#1056): If count items == max_batch, should we define the next batch immediately?
 
-	return &spb.DefineRevisionsResponse{OutstandingRevisions: outstanding}, nil
+	return resp, nil
+}
+
+// GetUnappliedRevisions returns a range of outstanding revisions for processing.
+func (s *Server) GetUnappliedRevisions(ctx context.Context,
+	in *spb.GetUnappliedRevisionsRequest) (*spb.GetUnappliedRevisionsResponse, error) {
+	root, high, err := s.getRootAndRev(ctx, in.DirectoryId)
+	if err != nil {
+		return nil, err
+	}
+	revs := &spb.GetUnappliedRevisionsResponse{
+		Begin: int64(root.Revision) + 1,
+		End:   high,
+	}
+	if revs.End < revs.Begin { // In case there is a race.
+		revs.End = revs.Begin // Avoid ugly results.
+	}
+	return revs, nil
+}
+
+// getRootAndRev returns the latest map root, and the first unprocessed
```

Comment:
The name is a bit confusing -- maybe because I made it do something confusing?
Perhaps this should be split into two functions?

LatestAppliedMapRevision
LatestDefinedBatchRevision

Reply:
Not relevant after another refactoring.